tools/mo/openvino/tools/mo/back/offline_transformations.py

```diff
@@ -37,10 +37,53 @@ def compress_model(func: object):
     from openvino.offline_transformations_pybind import compress_model_transformation  # pylint: disable=import-error,no-name-in-module
     compress_model_transformation(func)
 
-def apply_offline_transformations(input_model: str, framework: str, transforms: list, compress_fp16=False):
+
+def add_layouts(ov_function, argv: argparse.Namespace):
+    from openvino.runtime import Layout  # pylint: disable=import-error,no-name-in-module
+    from openvino.preprocess import PrePostProcessor  # pylint: disable=no-name-in-module,import-error
+
+    prep = PrePostProcessor(ov_function)
+    layout_values = argv.layout_values
+    if '' in layout_values:
+        if len(ov_function.inputs) > 1:
+            input_names = [list(ov_input.get_tensor().get_names())[0] for ov_input in ov_function.inputs]
+            raise Error('Layout without name can be specified for models with only one input, '
+                        'but provided model has {} inputs: \'{}\'. '
+                        'Please specify explicitly input/output name for --layout option'
+                        .format(len(input_names), input_names))
+        layout_values = {
+            list(ov_function.input().get_tensor().get_names())[0]: {
+                'source_layout': layout_values[''].get('source_layout'),
+                'target_layout': layout_values[''].get('target_layout')
+            }
+        }
+    set_layout_names = set(layout_values.keys())
+    for idx, ov_input in enumerate(ov_function.inputs):
+        found = set.intersection(set(ov_input.get_tensor().get_names()), set_layout_names)
+        if len(found) == 1:
+            node_name = list(found)[0]
+            found_layout = layout_values[node_name]
+            if found_layout['source_layout']:
+                prep.input(node_name).network().set_layout(Layout(found_layout['source_layout']))
+            if found_layout['target_layout']:
+                prep.input(node_name).tensor().set_layout(Layout(found_layout['target_layout']))
+
```

Comment:
should we have a check when there is more than 1 or zero intersections and output error message about incorrect input layout data?